hist/histpainter/src/THistPainter.cxx

```diff
@@ -7079,7 +7079,21 @@ Int_t THistPainter::PaintInit()
       }
       if (Hparam.xlowedge <=0 ) {
          if (Hoption.Same) {
-            Hparam.xlowedge = TMath::Power(10, gPad->GetUxmin());
+            TH1* h1 = nullptr;
+            TObject *obj;
+            TIter next(gPad->GetListOfPrimitives());
+            while ((obj = (TObject *)next())) {
```

Comment:
Should `TGraph2D` and `TF1` be in this list ?

Reply:
Yes .. that too... I will check